### DISCUSSION TITLE
fetch only once and always on mount

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#4756] [Q&A]
I have a comp that keeps fetching events , as the calls are heavy id like to only fetch the queries once unless the user moves away from the view.. in that case i want to serve from cache but also re fetch and refresh

I have the following set up:

{
    staleTime: 10 * (60 * 1000), // 10 mins
      cacheTime: 15 * (60 * 1000), // 15 mins
      refetchOnMount: 'always'
}

The first part works and the queries never re fetch (no network call) however it doesn't re fetch on remount...

Any help would be appreciated....

**Comment by TkDodo** [comment]
This should work - it's exactly what `refetchOnMount: 'always'` is for. Can you show a codesandbox reproduction where it doesn't work?

**Reply by cubecleveland** [reply]
HI Tk, yes. i will try an assemble this together. stay tuned, thank for your response. btw using version 3.34.2 should be supported , correct ?

**Reply by cubecleveland** [reply]
. Issue closed. for what its worth, the bug was on our end, if anyone else comes along this issue: we had an enabled condition on the query that was set to disabled the first time the comp was mounting so the 'always'; condition never triggered to re fetch the query.
